drivers/TableCRC.cpp

```diff
@@ -107,7 +107,7 @@ uint16_t Table_CRC_16bit_IBM[256] = {
     0x220,  0x8225, 0x822f, 0x22a,  0x823b, 0x23e,  0x234,  0x8231, 0x8213, 0x216,  0x21c,  0x8219
 };
 
-uint32_t Table_CRC_32bit_ANSI[256] = {
+extern const uint32_t Table_CRC_32bit_ANSI[256] = {
```

Comment:
extern is useless here as it's definition, external linkage by default .

One question related, you have sizes in definition and declaration. You could set these to  a macro, or a declaration does not need the size, thus if t his changes, no need to change the declaration neither `extern const uint32_t Table_CRC_32bit_ANSI[];`.

Reply:
external linkage is default in C, but with C++ if array constant is inside the namespace default linkage is internal.
I will replace 256 to a macro, but will keep the size in declaration as well.